### PR TITLE
Tickets/dm 34085

### PR DESCRIPTION
**Pull request by iagaponenko**

### CODE REVIEW

File: src/replica/HttpIngestTransModule.cc [hunk 1]
```diff
@@ -257,124 +270,179 @@ json HttpIngestTransModule::_beginTransaction() {
...
-    // return errors to clients).
-    try {
-        auto const config = controller()->serviceProvider()->config();
-        auto const databaseServices = controller()->serviceProvider()->databaseServices();
+    auto const config = controller()->serviceProvider()->config();
+    auto const databaseServices = controller()->serviceProvider()->databaseServices();
 
-        id = stoul(params().at("id"));
-        abort = query().requiredBool("abort");
+    TransactionId const id = stoul(params().at("id"));
```
Comment (n8pease) [inline src/replica/HttpIngestTransModule.cc:276]: any possible conversion error is handled above, or known to not happen at this point?

**Comment by n8pease** [comment]
I'm done reviewing, I checked the commits that do not say "web dashboard" or "web UI framework"

File: src/replica/DatabaseServices.h [hunk 2]
```diff
@@ -198,51 +201,105 @@ class JobInfo {
...
-    State state = State::STARTED;
+    /// The current state of the transaction
+    State state = State::IS_STARTING;
+
+    uint64_t beginTime = 0;         ///< The timestamp (milliseconds) when it was created (IS_STARTING).
+    uint64_t startTime = 0;         ///< The timestamp (milliseconds) when it started (STARTED).
+    uint64_t transitionTime = 0;    ///< The timestamp (milliseconds) when a transition
+                                    ///  commit (IS_FINISHING) / abort (IS_ABORTING) was initiated.
+    uint64_t endTime = 0;           ///< The timestamp (milliseconds) when start (START_FAILED),
+                                    ///  commit (FINISH_FAILED) or aborted (ABORT_FAILED) failed.
```
[the remark this answers is not in the document]
Comment (iagaponenko) [reply src/replica/DatabaseServices.h:246]: Good suggestion! I will fix the comment.

File: src/replica/Job.cc [hunk 3]
```diff
@@ -173,6 +186,23 @@ void Job::wait() {
...
+void Job::wait(chrono::milliseconds const& ivalMsec, WaitMonitorFunc const& func) {
+    string const context_ = context() + string(__func__) + " ";
+    LOGS(_log, LOG_LVL_DEBUG, context_);
+    if (_finished) return;
+    if (ivalMsec.count() == 0) throw invalid_argument(context_ + "callaback interval can't be 0.");
+    if (func == nullptr) throw invalid_argument(context_ + "callaback function not provided.");
+    auto const self = shared_from_this();
+    unique_lock<mutex> onFinishLock(_onFinishMtx);
+    while (!_onFinishCv.wait_for(onFinishLock, ivalMsec, [&] { return _finished.load(); })) {
+        // This is needed to prevent the deadlock in case if the called function
```
[the remark this answers is not in the document]
Comment (iagaponenko) [reply src/replica/Job.cc:198]: Goof suggestion! I will fix it.

**Pull request merged**